examples/react/start-workos/app/authkit/ssr/session.ts

```diff
@@ -0,0 +1,266 @@
+import { getWorkOS } from './workos';
+import type { AccessToken, AuthenticationResponse } from '@workos-inc/node';
+import type { AuthkitOptions, AuthkitResponse, CookieOptions, GetAuthURLOptions, Session } from './interfaces';
+
+const sessionHeaderName = 'x-workos-session';
+const middlewareHeaderName = 'x-workos-middleware';
+
+export function getAuthorizationUrl(options: GetAuthURLOptions = {}) {
+  const { returnPathname, screenHint, redirectUri } = options;
+
+  return getWorkOS().userManagement.getAuthorizationUrl({
+    provider: 'authkit',
+    clientId: getConfig('clientId'),
+    redirectUri: redirectUri || getConfig('redirectUri'),
+    state: returnPathname ? btoa(JSON.stringify({ returnPathname })) : undefined,
+    screenHint,
+  });
+}
+
+export function serializeCookie(name: string, value: string, options: Partial<CookieOptions> = {}): string {
+  const {
+    path = '/',
+    maxAge = getConfig('cookieMaxAge'),
+    secure = options.sameSite === 'none' ? true : getConfig('redirectUri').startsWith('https:'),
+    sameSite = 'lax',
+    domain = getConfig('cookieDomain'),
+  } = options;
+
+  let cookie = `${name}=${encodeURIComponent(value)}; Path=${path}; sameSite=${sameSite}; HttpOnly`;
+  cookie += `; Max-Age=${maxAge}`;
+  if (!maxAge) cookie += `; Expires=${new Date(0).toUTCString()}`;
+  if (secure) cookie += '; Secure';
+  if (domain) cookie += `; Domain=${domain}`;
+
+  return cookie;
+}
+
+export async function decryptSession(encryptedSession: string): Promise<Session> {
+  const cookiePassword = getConfig('cookiePassword');
+  return unsealData<Session>(encryptedSession, {
+    password: cookiePassword,
+  });
+}
+
+export async function encryptSession(session: Session) {
+  return sealData(session, {
+    password: getConfig('cookiePassword'),
+    ttl: 0,
+  });
+}
+
+export async function withAuth() {
+  const session = await getSessionFromCookie();
+
+  if (!session?.user) {
+    return { user: null };
+  }
+
+  const {
+    sid: sessionId,
+    org_id: organizationId,
+    role,
+    permissions,
+    entitlements,
+  } = decodeJwt<AccessToken>(session.accessToken);
+
+  return {
+    sessionId,
+    user: session.user,
+    organizationId,
+    role,
+    permissions,
+    entitlements,
+    impersonator: session.impersonator,
+    accessToken: session.accessToken,
+  };
+}
+
+export async function getSessionFromCookie() {
+  const cookieName = getConfig('cookieName') || 'wos-session';
+  const cookie = getCookie(cookieName);
+
+  if (cookie) {
+    return decryptSession(cookie);
+  }
+}
+
+export async function saveSession(sessionOrResponse: Session | AuthenticationResponse): Promise<void> {
+  const cookieName = getConfig('cookieName') || 'wos-session';
+  const encryptedSession = await encryptSession(sessionOrResponse);
+  setCookie(cookieName, encryptedSession);
+}
+
+const JWKS = lazy(() => createRemoteJWKSet(new URL(getWorkOS().userManagement.getJwksUrl(getConfig('clientId')))));
```

Comment:
Could you perhaps why `createRemoteJWTSet` is being wrapped in the `lazy` util (as a comment)?